### PR TITLE
Update `ember-cli` to `2.8.0`. Closes #152.

### PR DESCRIPTION
**Pull request by btecu**

### CODE REVIEW

**Comment by btecu** [comment]
Done

**Comment by kellyselden** [comment]
Test failures seem legit.

**Comment by btecu** [comment]
I removed legacy `containerDebugAdapter` that was needed for `Ember` prior `1.5.0`.

File: .travis.yml [hunk 1]
```diff
@@ -10,11 +10,8 @@ cache:
     - node_modules
 
 env:
-  - EMBER_TRY_SCENARIO=ember-1.10
```
Comment (nathanhammond) [inline .travis.yml]: Can we make this match:
https://github.com/ember-cli/ember-cli/blob/74562a4/blueprints/addon/files/.travis.yml#L15-L19

Comment (btecu) [inline .travis.yml]: Done

**Comment by nathanhammond** [comment]
Need a review from @rwjblue regarding the drop of `container-debug-adapter` and as a consequence what version we should release this change at.

**Comment by rwjblue** [comment]
Added to my queue, will try to review this weekend...

File: .travis.yml [hunk 2]
```diff
@@ -10,11 +10,8 @@ cache:
...
-  - EMBER_TRY_SCENARIO=ember-1.11
-  - EMBER_TRY_SCENARIO=ember-1.12
   - EMBER_TRY_SCENARIO=ember-1.13
-  - EMBER_TRY_SCENARIO=ember-2.0
+  - EMBER_TRY_SCENARIO=ember-lts-2.4
```
Comment (mixonic) [inline .travis.yml]: /cc @btecu

Comment (btecu) [inline .travis.yml]: @mixonic good catch. I've added the missing scenario.

File: package.json [hunk 3]
```diff
@@ -20,30 +20,31 @@
...
+    "ember-cli-test-loader": "^1.1.0",
     "ember-cli-uglify": "^1.2.0",
-    "ember-disable-prototype-extensions": "^1.0.0",
-    "ember-disable-proxy-controllers": "^1.0.0",
-    "ember-export-application-global": "^1.0.4",
-    "ember-try": "~0.0.8"
+    "ember-disable-prototype-extensions": "^1.1.0",
+    "ember-export-application-global": "^1.0.5",
+    "ember-load-initializers": "^0.5.1",
+    "ember-resolver": "^2.0.3",
```
Comment (mixonic) [inline package.json]: Will require https://github.com/ember-cli/ember-cli/pull/6254 to land before the addon will build. looking for a work around.

Comment (mixonic) [inline package.json]: PR in https://github.com/btecu/ember-resolver/pull/1!

**Comment by mixonic** [comment]
@rwjblue this is seeming gtg, but you should review.

**Comment by rwjblue** [comment]
I'm somewhat concerned with the removal of the container debug adapter. Before release we should move the adapter to a separate folder and only merge it in if the app is using < 1.7. This would keep us backwards compatible but still allow us to remove that dead code for recent Ember versions.

**Comment by rwjblue** [comment]
@mixonic - if this is blocking your work, I can land this now and just block releasing for that last concern. If you aren't blocked, I'd prefer to update before landing...

**Comment by btecu** [comment]
`ContainerDebugAdapter` was added in ember in `1.5`. I can't think of anything blocking a user from upgrading ember to `1.5` or later from versions `1.0` - `1.4`.

**Comment by rwjblue** [comment]
Sure, I completely agree with you! Sadly, that isn't how SemVer works. As of right now we support Ember 1.0+ (with inspector support), if we change that we have to do a major version bump. I'd prefer to avoid bumping major versions for this, because:

* It doesn't seem necessary (I laid out a path above that wouldn't require it).
* The coming module unification work may require another major version update and I don't want to expend our "user goodwill" with two major bumps in close proximity.

**Comment by nathanhammond** [comment]
Is there any particular reason to tie the CLI 2.8 update to the removal of `containerDebugAdapter`? I'm not sure why they're both in this PR.

**Comment by btecu** [comment]
@nathanhammond the tests for `containerDebugAdapter` were failing.

**Comment by btecu** [comment]
@rwjblue since this is only updating `ember-cli` and removing `ContainerDebugAdapter`, it can be left open until your model unification changes are ready, to avoid more than one major version change.

cc @iezer

**Comment by iezer** [comment]
@btecu Actually there's an easy fix for those tests. I think they are failing because a transpilation failure/naming clash of the name `module`. This is the fix. https://github.com/ember-cli/ember-resolver/commit/4d8fda96b79f4164abeb661668b0d18ab642cdfa  I can submit a PR to your branch.

**Comment by rwjblue** [comment]
@btecu - Based on @teddyzeenny's comment in https://github.com/ember-cli/ember-resolver/pull/156#discussion_r82121794, the inspector continues to require `ContainerDebugAdapter` to be able to function.  We cannot remove it.  The comment there was specifically related to the guard for when `Ember.ContainerDebugAdapter` is *not* present.

**Comment by btecu** [comment]
@rwjblue If the ember-resolver version has a major change, the inspector could still use the old one until it decides to stop supporting older Ember versions, right?

**Comment by iezer** [comment]
@btecu I opened a PR that fixes the tests https://github.com/btecu/ember-resolver/pull/2

**Comment by btecu** [comment]
@iezer that might work as well.

**Comment by rwjblue** [comment]
@btecu - I'm sorry, I don't think I'm being clear.  The inspector **requires** this file, it is completely unrelated to Ember version.  The inspector must find a `container-debug-adapter:main` or it will not be able to function properly.  This isn't just for older Ember version support, it is required for current versions of Ember.

**Comment by btecu** [comment]
@iezer I cherrypicked those fixes.
@rwjblue I updated the PR to not remove the `ContainerDebugAdapter`.

**Comment by rwjblue** [comment]
Thank you @btecu and @iezer!

**Pull request merged**